src/develop/imageop.h

```diff
@@ -378,6 +382,13 @@ typedef struct dt_iop_module_t
   void (*process_tiling)(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t *piece, void *i, void *o,
                          const struct dt_iop_roi_t *roi_in, const struct dt_iop_roi_t *roi_out,
                          const int bpp);
+  /** NOTE: most likely, you do not want to implement those ^ two functions, but next two: */
+  /** the simplest variant of process(). you can only use OpeMP SIMD here, no intrinsics */
```

Comment:
a crazy lot of code. cool stuff, i'm sure it introduces subtle bugs :)

so the modules you didn't touch yet don't do anything at all? if we were just treating process() as process_sse() because it potentially contains sse? then it would just fail for the case that the user explicitly requested omp simd/plain or has limited cpu capabilities? or how does that currently work with the three versions of process()?